### PR TITLE
BinaryReaderIR: set name of type if given in name section

### LINKED ISSUE #1687: Type identifiers in the name section are lost during round trip.
When converting a binary that contains an extended name section to WAT, the identifiers are all preserved, except type and label identifiers, which are seemingly just ignored.

### PR DESCRIPTION
**Pull request by keithw**
Fixes #1687

### CODE REVIEW

**Comment by sbc100** [comment]
Add a test?   I guess anything test that round-trips through IR with a named types would do?

**Comment by keithw** [comment]
Done.

File: test/roundtrip/named-type.txt [hunk 1]
```diff
@@ -0,0 +1,8 @@
+;;; TOOL: run-roundtrip
+;;; ARGS: --stdout --debug-names
+(module $a
+  (type $b (func)))
+(;; STDOUT ;;;
+(module $a
+  (type $b (func)))
```
Comment (sbc100) [inline test/roundtrip/named-type.txt:7]: Sorry, can we just update the existing `test/roundtrip/debug-names.txt` test?

Comment (keithw) [reply test/roundtrip/named-type.txt]: Sure, how about this?

**Review by sbc100 — APPROVED** [review]

**Pull request merged**